Android Source/mobile/src/main/java/de/jo2k/loggingmodule/DataHandler.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Basic implementation of a simple logger that writes a JSON stream to the mobile's external storage.
+ * JSON might be non-trivial to parse, but the data structure that can be generated per sensor
+ * are not uniform so it is probably best to stay flexible.
+ *
+ * Contributed by Jonathan Liebers (c) 2021, git@jo2k.de
+ *
+ * Thanks to `Steppschuh` (Stephan Schultz) for the great initial work.
+ *
+ * License: Apache License 2.0
+ */
+
+package de.jo2k.loggingmodule;
+
+import android.os.Environment;
+import android.util.Log;
+
+import net.steppschuh.datalogger.data.DataBatch;
+import net.steppschuh.sensordatalogger.BuildConfig;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import static net.steppschuh.sensordatalogger.PhoneApp.TAG;
+
+public class DataHandler {
+    private boolean isRecording = false;
+    private FileWriter filewriter;
+    private String filename;
+    private boolean firstEntryWasWritten;
+
+    public void logDataBatch(DataBatch dataBatch, String sourceNodeId) {
+        /**
+         * Writes newly obtained data to the logfile.
+         * User should call DataHandler.startRecording() beforehand to initialize a file.
+         * Returns silently if no recording was started beforehand.
+         */
+        if (!isRecording)
+            return;
+
+        // Log.v(TAG, sourceNodeId);
+        // Log.v(TAG, dataBatch.toString());
+
+        try {
+            // write the comma only if we added some entries before
+            if (firstEntryWasWritten) {
+                filewriter.write(",");
+            }
+            firstEntryWasWritten = true;
+            filewriter.write(dataBatch.toJson());
+        } catch (IOException e) {
+            Log.e(TAG, "FileWriter could not write.");
+            e.printStackTrace();
+        }
+    }
+
+    public boolean isRecording() {
+        /**
+         * Returns the status of the DataHandler, true if a recording is in progress, false otherwise.
+         */
+        return isRecording;
+    }
+
+    public void startRecording(String filename) {
```

Comment:
I would change the parameter to accept a `File` here. This way you can pass it from the Activity, which has the context to use `Context.getExternalFilesDir(String)`.

Reply:
:+1: